### ISSUE TITLE
invalid conversion from ‘caffe2::Blob*’ to ‘uint32_t {aka unsigned int}’

### ISSUE THREAD

**Issue by foreignmeloman** [pytorch/pytorch#37822]
Hi! I've encountered a strange error while building PyTorch 1.4.0 on RHEL 6 machine using gcc 6.2.1, which seems like a casting issue.

Am I doing something wrong here?

My environment is:
```
export BUILD_TEST=0
export USE_CUDA=0
export USE_NNPACK=0
export USE_OPENMP=0
export USE_MPI=OFF
export USE_NCCL=OFF
export USE_NUMA=OFF
export USE_MKLDNN=0
```
Build log tail:
```
...
[ 26%] Built target Caffe2_perfkernels_avx2
[ 26%] Built target Caffe2_perfkernels_avx512
[ 26%] Built target Caffe2_perfkernels_avx
[ 27%] Built target caffe2_dnnlowp_avx2_ops
[ 27%] Building CXX object caffe2/CMakeFiles/torch.dir/opt/onnxifi_op.cc.o
In file included from /tmp/pytorch-1.4.0/c10/core/CPUAllocator.h:7:0,
                 from /tmp/pytorch-1.4.0/caffe2/core/allocator.h:3,
                 from /tmp/pytorch-1.4.0/caffe2/core/context.h:9,
                 from /tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.h:8,
                 from /tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.cc:1:
/tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.h: In instantiation of ‘caffe2::OnnxifiOp<Context>::buildBackendAndGraph(caffe2::Workspace*, const std::vector<long unsigned int, std::allocator<long unsigned int> >&, const string&)::<lambda()> [with Context = caffe2::CPUContext]’:
/tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.h:185:22:   required from ‘struct caffe2::OnnxifiOp<Context>::buildBackendAndGraph(caffe2::Workspace*, const std::vector<long unsigned int, std::allocator<long unsigned int> >&, const string&) [with Context = caffe2::CPUContext; std::string = std::basic_string<char>]::<lambda()>’
/tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.h:264:5:   required from ‘void caffe2::OnnxifiOp<Context>::buildBackendAndGraph(caffe2::Workspace*, const std::vector<long unsigned int, std::allocator<long unsigned int> >&, const string&) [with Context = caffe2::CPUContext; std::string = std::basic_string<char>]’
/tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.h:124:25:   required from ‘caffe2::OnnxifiOp<Context>::OnnxifiOp(const caffe2::OperatorDef&, caffe2::Workspace*) [with Context = caffe2::CPUContext]’
/tmp/pytorch-1.4.0/c10/util/Registry.h:184:26:   required from ‘static ObjectPtrType c10::Registerer<SrcType, ObjectPtrType, Args>::DefaultCreator(Args ...) [with DerivedType = caffe2::OnnxifiOp<caffe2::CPUContext>; SrcType = std::basic_string<char>; ObjectPtrType = std::unique_ptr<caffe2::OperatorBase>; Args = {const caffe2::OperatorDef&, caffe2::Workspace*}]’
/tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.cc:436:1:   required from here
/tmp/pytorch-1.4.0/c10/util/Logging.h:239:33: error: invalid conversion from ‘caffe2::Blob*’ to ‘uint32_t {aka unsigned int}’ [-fpermissive]
   CAFFE_ENFORCE_THAT_IMPL(Equals((x), (y)), #x " == " #y, __VA_ARGS__)
                                 ^
/tmp/pytorch-1.4.0/c10/util/Logging.h:207:61: note: in definition of macro ‘CAFFE_ENFORCE_THAT_IMPL’
     const EnforceFailMessage& CAFFE_ENFORCE_THAT_IMPL_r_ = (condition); \
                                                             ^~~~~~~~~
/tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.h:250:7: note: in expansion of macro ‘CAFFE_ENFORCE_EQ’
       CAFFE_ENFORCE_EQ(
       ^~~~~~~~~~~~~~~~
/tmp/pytorch-1.4.0/c10/util/Logging.h:239:33: error: too few arguments to function
   CAFFE_ENFORCE_THAT_IMPL(Equals((x), (y)), #x " == " #y, __VA_ARGS__)
                                 ^
/tmp/pytorch-1.4.0/c10/util/Logging.h:207:61: note: in definition of macro ‘CAFFE_ENFORCE_THAT_IMPL’
     const EnforceFailMessage& CAFFE_ENFORCE_THAT_IMPL_r_ = (condition); \
                                                             ^~~~~~~~~
/tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.h:250:7: note: in expansion of macro ‘CAFFE_ENFORCE_EQ’
       CAFFE_ENFORCE_EQ(
       ^~~~~~~~~~~~~~~~
In file included from /opt/rh/devtoolset-6/root/usr/include/c++/6.2.1/memory:79:0,
                 from /tmp/protobuf-3.8.0/src/google/protobuf/stubs/common.h:41,
                 from /tmp/protobuf-3.8.0/src/google/protobuf/io/coded_stream.h:135,
                 from /tmp/pytorch-1.4.0/build/third_party/onnx/onnx/onnx_onnx_torch-ml.pb.h:23,
                 from /tmp/pytorch-1.4.0/build/third_party/onnx/onnx/onnx-ml.pb.h:2,
                 from /tmp/pytorch-1.4.0/third_party/onnx/onnx/onnx_pb.h:50,
                 from /tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.h:5,
                 from /tmp/pytorch-1.4.0/caffe2/opt/onnxifi_op.cc:1:
/opt/rh/devtoolset-6/root/usr/include/c++/6.2.1/functional:2115:7: error: ‘std::function<_Res(_ArgTypes ...)>::function(_Functor) [with _Functor = caffe2::OnnxifiOp<Context>::buildBackendAndGraph(caffe2::Workspace*, const std::vector<long unsigned int, std::allocator<long unsigned int> >&, const string&) [with Context = caffe2::CPUContext; std::string = std::basic_string<char>]::<lambda()>; <template-parameter-2-2> = void; <template-parameter-2-3> = void; _Res = std::shared_ptr<caffe2::onnx::BackendGraphInfo>; _ArgTypes = {}]’, declared using local type ‘caffe2::OnnxifiOp<Context>::buildBackendAndGraph(caffe2::Workspace*, const std::vector<long unsigned int, std::allocator<long unsigned int> >&, const string&) [with Context = caffe2::CPUContext; std::string = std::basic_string<char>]::<lambda()>’, is used but never defined [-fpermissive]
       function<_Res(_ArgTypes...)>::
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
caffe2/CMakeFiles/torch.dir/build.make:17819: recipe for target 'caffe2/CMakeFiles/torch.dir/opt/onnxifi_op.cc.o' failed
gmake[2]: *** [caffe2/CMakeFiles/torch.dir/opt/onnxifi_op.cc.o] Error 1
CMakeFiles/Makefile2:1741: recipe for target 'caffe2/CMakeFiles/torch.dir/all' failed
gmake[1]: *** [caffe2/CMakeFiles/torch.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
gmake: *** [all] Error 2
Building wheel torch-1.4.0a0+6f51ff4
-- Building version 1.4.0a0+6f51ff4
cmake --build . --target install --config Release -- -j 1
Traceback (most recent call last):
  File "setup.py", line 755, in <module>
    build_deps()
  File "setup.py", line 316, in build_deps
    cmake=cmake)
  File "/tmp/pytorch-1.4.0/tools/build_pytorch_libs.py", line 62, in build_caffe2
    cmake.build(my_env)
  File "/tmp/pytorch-1.4.0/tools/setup_helpers/cmake.py", line 335, in build
    self.run(build_args, my_env)
  File "/tmp/pytorch-1.4.0/tools/setup_helpers/cmake.py", line 141, in run
    check_call(command, cwd=self.build_dir, env=env)
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install', '--config', 'Release', '--', '-j', '1']' returned non-zero exit status 2.

```

cc @malfet

**Comment by mruberry** [comment]
Sorry to hear you're having trouble building PyTorch. What happens if you try 1.5?

**Comment by foreignmeloman** [comment]
I've tried to build 1.5.0, the issue is gone. Thanks!

**Comment by foreignmeloman** [comment]
FYI This issue seems to be fixed in #31602 as backporting the change from 1.5.0 solved the issue.
